src/codemagic/models/provisioning_profile.py

```diff
@@ -91,7 +92,8 @@ def provisions_all_devices(self) -> bool:
 
     @property
     def application_identifier(self) -> str:
-        return self._plist.get('Entitlements', dict()).get('application-identifier', '')
+        entitlements = self._plist.get('Entitlements', dict())
+        return next((entitlements[key] for key in entitlements.keys() if 'application-identifier' in key), '')
```

Comment:
Simliarly to `has_beta_entitlements` suggestion:
```suggestion
        for key, value in entitlements.items():
            if 'application-identifier' in key:
                return value
        return ''
```

Reply:
fixed in d7de32a87dd4bfd264c78c390a182d62a95526cd